### PR TITLE
Migrate to MSBuild

### PR DESCRIPTION
**Pull request by dougbu**
Rebase nimullen/migration branch from PR #5749 and carry on from there. See descriptions of individual commits for more detail.

Current status: Command-line builds work cleanly. Mvc.NoFun.sln and Mvc.OnlyFun.sln generally load, build, and test cleanly in Visual Studio. Still experiencing some "Non responsive" periods and slow loads, especially if solution is loaded at same time as VS is started. Mvc.sln still tends to hang VS.

Will file a few more bugs (mostly around C# editing performance and testing experience) and add them to #5482. Have already checked off fixed bugs in #5482.

### CODE REVIEW

**Comment by dougbu** [comment]
@natemcmaster @pranavkm @Eilon please have a look at this instead of #5749. But note pushing remains blocked on VS issues.

File: src/Microsoft.AspNetCore.Mvc.Core/Microsoft.AspNetCore.Mvc.Core.csproj [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    <ProjectReference Include="..\Microsoft.AspNetCore.Mvc.Abstractions\Microsoft.AspNetCore.Mvc.Abstractions.csproj" />
+
+    <PackageReference Include="Microsoft.AspNetCore.Authorization" Version="1.2.0-*" />
+    <PackageReference Include="Microsoft.AspNetCore.Hosting.Abstractions" Version="1.2.0-*" />
+    <PackageReference Include="Microsoft.AspNetCore.Http" Version="1.2.0-*" />
+    <PackageReference Include="Microsoft.AspNetCore.ResponseCaching.Abstractions" Version="1.2.0-*" />
+    <PackageReference Include="Microsoft.AspNetCore.Routing" Version="1.2.0-*" />
+    <PackageReference Include="Microsoft.AspNetCore.Routing.DecisionTree.Sources" Version="1.2.0-*" PrivateAssets="All" />
+    <PackageReference Include="Microsoft.Extensions.ClosedGenericMatcher.Sources" Version="1.2.0-*" PrivateAssets="All" />
+    <PackageReference Include="Microsoft.Extensions.DependencyModel" Version="1.1.0-*" />
```
Comment (dougbu) [inline src/Microsoft.AspNetCore.Mvc.Core/Microsoft.AspNetCore.Mvc.Core.csproj]: Is this the correct version of this package?

File: samples/MvcSandbox/MvcSandbox.csproj [hunk 2]
```diff
@@ -0,0 +1,27 @@
+<Project Sdk="Microsoft.NET.Sdk.Web">
+
+  <Import Project="..\..\build\common.props" />
+
+  <PropertyGroup>
+    <TargetFrameworks>net452;netcoreapp1.1</TargetFrameworks>
+    <TargetFrameworks Condition="'$(OS)' != 'Windows_NT'">netcoreapp1.1</TargetFrameworks>
+    <RuntimeIdentifier Condition=" '$(TargetFramework)' != 'netcoreapp1.1' ">win7-x64</RuntimeIdentifier>
```
Comment (dougbu) [inline samples/MvcSandbox/MvcSandbox.csproj]: I hesitate to remove this because confirming everything works in command-line and VS runs as well as our functional tests might miss something. Have at least confirmed the Microsoft.NET.Sdk.Web targets do *not* contain anything similar.

Anyone know if these lines in the web site project files can be removed?

File: test/Microsoft.AspNetCore.Mvc.Core.Test/Microsoft.AspNetCore.Mvc.Core.Test.csproj [hunk 3]
```diff
@@ -0,0 +1,39 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+  <Import Project="..\..\build\common.props" />
+
+  <PropertyGroup>
+    <TargetFrameworks>netcoreapp1.1;net452</TargetFrameworks>
+    <TargetFrameworks Condition="'$(OS)' != 'Windows_NT'">netcoreapp1.1</TargetFrameworks>
+    <PreserveCompilationContext>true</PreserveCompilationContext>
```
Comment (dougbu) [inline test/Microsoft.AspNetCore.Mvc.Core.Test/Microsoft.AspNetCore.Mvc.Core.Test.csproj]: Was present in may test project.json files. But, was actually needed only in 2 test projects.

File: test/Microsoft.AspNetCore.Mvc.Razor.Test/Internal/ViewComponentTagHelperDescriptorFactoryTest.cs [hunk 4]
```diff
@@ -87,6 +86,13 @@ public static TheoryData TypeData
         [MemberData(nameof(TypeData))]
         public void GetCSharpTypeName_ReturnsCorrectTypeNames(Type type, string expected)
         {
+            // Arrange
+            // Work around problems serializing / deserializing this ref type.
+            if (type == null)
+            {
+                type = typeof(ViewComponentTagHelperDescriptorFactoryTest).GetMethod("MethodWithOutParam").GetParameters().First().ParameterType;
+            }
```
Comment (dougbu) [inline test/Microsoft.AspNetCore.Mvc.Razor.Test/Internal/ViewComponentTagHelperDescriptorFactoryTest.cs]: @ajaybhargavb did you file a bug about the serialization / deserialization problems with the `out List<char*[]>` parameter's `Type`? Would like to reference it here.

Comment (dougbu) [inline test/Microsoft.AspNetCore.Mvc.Razor.Test/Internal/ViewComponentTagHelperDescriptorFactoryTest.cs]: BTW I suspect the issue is actually in serialization -- ends up as an open generic `List<>` type.

File: src/Microsoft.AspNetCore.Mvc.Core/Microsoft.AspNetCore.Mvc.Core.csproj (hunk 1 above)
Comment (natemcmaster) [inline src/Microsoft.AspNetCore.Mvc.Core/Microsoft.AspNetCore.Mvc.Core.csproj]: There is a 2.0.0 version being built...https://github.com/dotnet/core-setup/blob/master/src/Microsoft.Extensions.DependencyModel/. Let's add that to the list of things we upgrade when we move everything to netstandard2.0 / .NET Core 2.0. cref https://github.com/aspnet/Coherence/issues/166

File: test/Microsoft.AspNetCore.Mvc.Razor.Test/Internal/ViewComponentTagHelperDescriptorFactoryTest.cs (hunk 4 above)
Comment (ajaybhargavb) [inline test/Microsoft.AspNetCore.Mvc.Razor.Test/Internal/ViewComponentTagHelperDescriptorFactoryTest.cs]: Not yet. I was waiting for your PR to repro this issue. I'll link it here once I am done.

File: test/Microsoft.AspNetCore.Mvc.ApiExplorer.Test/Microsoft.AspNetCore.Mvc.ApiExplorer.Test.csproj [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+  <PropertyGroup>
+    <TargetFrameworks>netcoreapp1.1;net452</TargetFrameworks>
+    <TargetFrameworks Condition="'$(OS)' != 'Windows_NT'">netcoreapp1.1</TargetFrameworks>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <ProjectReference Include="..\..\src\Microsoft.AspNetCore.Mvc\Microsoft.AspNetCore.Mvc.csproj" />
+    <ProjectReference Include="..\Microsoft.AspNetCore.Mvc.TestCommon\Microsoft.AspNetCore.Mvc.TestCommon.csproj" />
+
+    <PackageReference Include="Microsoft.DotNet.InternalAbstractions" Version="1.0.0" />
```
Comment (natemcmaster) [inline test/Microsoft.AspNetCore.Mvc.ApiExplorer.Test/Microsoft.AspNetCore.Mvc.ApiExplorer.Test.csproj]: Can you try removing this? IIRC this was added because early builds of xunit needed it. This has since been fixed.

Unless, of course, this project uses API from that package. In which case, switch this to Microsoft.DotNet.PlatformAbstractions/1.1.0 instead.

File: test/Microsoft.AspNetCore.Mvc.Cors.Test/Microsoft.AspNetCore.Mvc.Cors.Test.csproj [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+    <TargetFrameworks>netcoreapp1.1;net452</TargetFrameworks>
+    <TargetFrameworks Condition="'$(OS)' != 'Windows_NT'">netcoreapp1.1</TargetFrameworks>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <ProjectReference Include="..\..\src\Microsoft.AspNetCore.Mvc.Cors\Microsoft.AspNetCore.Mvc.Cors.csproj" />
+    <ProjectReference Include="..\Microsoft.AspNetCore.Mvc.TestCommon\Microsoft.AspNetCore.Mvc.TestCommon.csproj" />
+
+    <PackageReference Include="Microsoft.AspNetCore.Http" Version="1.2.0-*" />
+    <PackageReference Include="Microsoft.DotNet.InternalAbstractions" Version="1.0.0" />
```
Comment (natemcmaster) [inline test/Microsoft.AspNetCore.Mvc.Cors.Test/Microsoft.AspNetCore.Mvc.Cors.Test.csproj]: Ditto...repeat for all test projects.

File: test/Microsoft.AspNetCore.Mvc.FunctionalTests/Microsoft.AspNetCore.Mvc.FunctionalTests.csproj [hunk 7]
```diff
@@ -0,0 +1,66 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <Import Project="..\..\build\common.props" />
+
+  <PropertyGroup>
+    <TargetFrameworks>netcoreapp1.1;</TargetFrameworks>
+    <DefineConstants>$(DefineConstants);__RemoveThisBitTo__GENERATE_BASELINES;FUNCTIONAL_TESTS</DefineConstants>
+    <PackageTargetFallback>$(PackageTargetFallback);portable-net451+win8</PackageTargetFallback>
```
Comment (natemcmaster) [inline test/Microsoft.AspNetCore.Mvc.FunctionalTests/Microsoft.AspNetCore.Mvc.FunctionalTests.csproj]: What are we using that still requires this?

File: test/Microsoft.AspNetCore.Mvc.FunctionalTests/Microsoft.AspNetCore.Mvc.FunctionalTests.csproj [hunk 8]
```diff
@@ -0,0 +1,66 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <Import Project="..\..\build\common.props" />
+
+  <PropertyGroup>
+    <TargetFrameworks>netcoreapp1.1;</TargetFrameworks>
```
Comment (natemcmaster) [inline test/Microsoft.AspNetCore.Mvc.FunctionalTests/Microsoft.AspNetCore.Mvc.FunctionalTests.csproj]: Nit: use the singular form. `<TargetFramework>netcoreapp1.1</TargetFramework>`

File: test/Microsoft.AspNetCore.Mvc.Razor.Test/Internal/ViewComponentTagHelperDescriptorFactoryTest.cs (hunk 4 above)
Comment (natemcmaster) [inline test/Microsoft.AspNetCore.Mvc.Razor.Test/Internal/ViewComponentTagHelperDescriptorFactoryTest.cs]: Is this bug introduced by converting to C# 7.0? We had a few other, minor problems when switching to the new compiler.

File: test/Microsoft.AspNetCore.Mvc.TestCommon/Microsoft.AspNetCore.Mvc.TestCommon.csproj [hunk 9]
```diff
@@ -0,0 +1,24 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <Import Project="..\..\build\common.props" />
+
+  <PropertyGroup>
+    <TargetFrameworks>net452;netstandard1.6</TargetFrameworks>
+    <TargetFrameworks Condition="'$(OS)' != 'Windows_NT'">netstandard1.6</TargetFrameworks>
```
Comment (natemcmaster) [inline test/Microsoft.AspNetCore.Mvc.TestCommon/Microsoft.AspNetCore.Mvc.TestCommon.csproj]: Probably not required, but doesn't hurt to leave it.

File: test/Microsoft.AspNetCore.Mvc.WebApiCompatShimTest/Microsoft.AspNetCore.Mvc.WebApiCompatShimTest.csproj [hunk 10]
```diff
@@ -0,0 +1,29 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <Import Project="..\..\build\common.props" />
+
+  <PropertyGroup>
+    <TargetFrameworks>netcoreapp1.1;net452</TargetFrameworks>
+    <TargetFrameworks Condition="'$(OS)' != 'Windows_NT'">netcoreapp1.1</TargetFrameworks>
+    <PackageTargetFallback Condition=" '$(TargetFramework)' == 'netcoreapp1.1' ">$(PackageTargetFallback);portable-net451+win8</PackageTargetFallback>
```
Comment (natemcmaster) [inline test/Microsoft.AspNetCore.Mvc.WebApiCompatShimTest/Microsoft.AspNetCore.Mvc.WebApiCompatShimTest.csproj]: Remove?

File: test/WebSites/ApiExplorerWebSite/ApiExplorerWebSite.csproj [hunk 11]
```diff
@@ -0,0 +1,21 @@
+<Project Sdk="Microsoft.NET.Sdk.Web">
+
+  <Import Project="..\..\..\build\common.props" />
+
+  <PropertyGroup>
+    <TargetFrameworks>net452;netcoreapp1.1</TargetFrameworks>
+    <TargetFrameworks Condition="'$(OS)' != 'Windows_NT'">netcoreapp1.1</TargetFrameworks>
+    <RuntimeIdentifier Condition=" '$(TargetFramework)' != 'netcoreapp1.1' ">win7-x64</RuntimeIdentifier>
```
Comment (natemcmaster) [inline test/WebSites/ApiExplorerWebSite/ApiExplorerWebSite.csproj]: Supposedly, we don't have to do this anymore. Last week the SDK remove the requirement to specify RuntimeIdentifier for .NET Framework projects. Can you try removing this and see if it still works? If not, it may be a bug in the SDK.

Repeat for all test websites below...

File: test/WebSites/ControllersFromServicesClassLibrary/ControllersFromServicesClassLibrary.csproj [hunk 12]
```diff
@@ -0,0 +1,14 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <Import Project="..\..\..\build\common.props" />
+
+  <PropertyGroup>
+    <TargetFrameworks>net452;netcoreapp1.1</TargetFrameworks>
```
Comment (natemcmaster) [inline test/WebSites/ControllersFromServicesClassLibrary/ControllersFromServicesClassLibrary.csproj]: Target .NET Standard? This is a class library project.

File: test/WebSites/FilesWebSite/FilesWebSite.csproj [hunk 13]
```diff
@@ -0,0 +1,28 @@
...
+
+  <PropertyGroup>
+    <TargetFrameworks>net452;netcoreapp1.1</TargetFrameworks>
+    <TargetFrameworks Condition="'$(OS)' != 'Windows_NT'">netcoreapp1.1</TargetFrameworks>
+    <RuntimeIdentifier Condition=" '$(TargetFramework)' != 'netcoreapp1.1' ">win7-x64</RuntimeIdentifier>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <EmbeddedResource Include="EmbeddedResources\**" Exclude="bin\**;obj\**;**\*.xproj;packages\**;@(EmbeddedResource)" />
+    <Content Update="sample.txt;web.config">
```
Comment (natemcmaster) [inline test/WebSites/FilesWebSite/FilesWebSite.csproj]: For sure don't need web.config here, might not need sample.txt. The SDK will pick up web.config by default.

File: test/WebSites/FilesWebSite/FilesWebSite.csproj [hunk 14]
```diff
@@ -0,0 +1,28 @@
...
+  <Import Project="..\..\..\build\common.props" />
+
+  <PropertyGroup>
+    <TargetFrameworks>net452;netcoreapp1.1</TargetFrameworks>
+    <TargetFrameworks Condition="'$(OS)' != 'Windows_NT'">netcoreapp1.1</TargetFrameworks>
+    <RuntimeIdentifier Condition=" '$(TargetFramework)' != 'netcoreapp1.1' ">win7-x64</RuntimeIdentifier>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <EmbeddedResource Include="EmbeddedResources\**" Exclude="bin\**;obj\**;**\*.xproj;packages\**;@(EmbeddedResource)" />
```
Comment (natemcmaster) [inline test/WebSites/FilesWebSite/FilesWebSite.csproj]: You can remove the Exclude attribute here as the "Include" wouldn't match up with anything in the Exclude patterns.

File: test/WebSites/RazorWebSite/RazorWebSite.csproj [hunk 15]
```diff
@@ -0,0 +1,33 @@
...
+    <ProjectReference Include="..\..\..\src\Microsoft.AspNetCore.Mvc\Microsoft.AspNetCore.Mvc.csproj" />
+    <ProjectReference Include="..\Microsoft.AspNetCore.Mvc.TestConfiguration\Microsoft.AspNetCore.Mvc.TestConfiguration.csproj" />
+
+    <PackageReference Include="Microsoft.AspNetCore.Server.IISIntegration" Version="1.2.0-*" />
+    <PackageReference Include="Microsoft.AspNetCore.Server.Kestrel" Version="1.2.0-*" />
+    <PackageReference Include="Microsoft.AspNetCore.StaticFiles" Version="1.2.0-*" />
+    <PackageReference Include="Microsoft.Extensions.FileProviders.Embedded" Version="1.2.0-*" />
+  </ItemGroup>
+
+  <PropertyGroup Condition=" '$(TargetFramework)' == 'net452' ">
```
Comment (natemcmaster) [inline test/WebSites/RazorWebSite/RazorWebSite.csproj]: Nit: keep all property groups together at the top of the top file.

Also, why are these necessary? The SDK already defines contants per-TFM.

File: test/WebSites/UserClassLibrary/UserClassLibrary.csproj [hunk 16]
```diff
@@ -0,0 +1,12 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <Import Project="..\..\..\build\common.props" />
+
+  <PropertyGroup>
+    <TargetFrameworks>net452;netstandard1.3</TargetFrameworks>
+    <TargetFrameworks Condition="'$(OS)' != 'Windows_NT'">netstandard1.3</TargetFrameworks>
+    <AssemblyName>UserClassLibrary</AssemblyName>
```
Comment (natemcmaster) [inline test/WebSites/UserClassLibrary/UserClassLibrary.csproj]: Nit: both AssemblyName and PackageId aren't required here. These are the default values.

File: build/repo.props [hunk 17]
```diff
@@ -0,0 +1,8 @@
+<Project>
+  <ItemGroup>
+    <ExcludeFromTest Include="$(RepositoryRoot)test\Microsoft.AspNetCore.Mvc.TestCommon\*.csproj" />
+    <!-- Exclude with and without extra slash in case DefaultItems.targets is fixed. -->
+    <ExcludeSolutions Include="$(RepositoryRoot)\Mvc.*Fun.sln" />
```
Comment (natemcmaster) [inline build/repo.props]: Oops. Open an issue on KoreBuild.

File: test/WebSites/ControllersFromServicesClassLibrary/ControllersFromServicesClassLibrary.csproj (hunk 12 above)
Comment (pranavkm) [inline test/WebSites/ControllersFromServicesClassLibrary/ControllersFromServicesClassLibrary.csproj]: Looks like this missing the `preserveCompilationContext` flag from the project.json. I think project.json required an executable to produce a deps file.

File: test/Microsoft.AspNetCore.Mvc.WebApiCompatShimTest/Microsoft.AspNetCore.Mvc.WebApiCompatShimTest.csproj (hunk 10 above)
Comment (pranavkm) [inline test/Microsoft.AspNetCore.Mvc.WebApiCompatShimTest/Microsoft.AspNetCore.Mvc.WebApiCompatShimTest.csproj]: Nope. WebAPIShim references portable packages.

File: test/Microsoft.AspNetCore.Mvc.Test/Microsoft.AspNetCore.Mvc.Test.csproj [hunk 18]
```diff
@@ -0,0 +1,31 @@
...
+    <PackageReference Include="Microsoft.Extensions.DependencyInjection" Version="1.2.0-*" />
+    <PackageReference Include="Microsoft.Extensions.Logging" Version="1.2.0-*" />
+    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="15.0.0-*" />
+    <PackageReference Include="Moq" Version="4.6.36-*" />
+    <PackageReference Include="xunit" Version="2.2.0-*" />
+    <PackageReference Include="xunit.runner.visualstudio" Version="2.2.0-*" />
+  </ItemGroup>
+
+  <ItemGroup Condition=" '$(TargetFramework)' == 'netcoreapp1.1' ">
+    <PackageReference Include="System.Xml.XmlDocument" Version="$(CoreFxVersion)-*" />
```
Comment (pranavkm) [inline test/Microsoft.AspNetCore.Mvc.Test/Microsoft.AspNetCore.Mvc.Test.csproj]: Is this still needed? The Formatter project should bring in this reference

File: test/Microsoft.AspNetCore.Mvc.Razor.Test/Internal/ViewComponentTagHelperDescriptorFactoryTest.cs (hunk 4 above)
Comment (pranavkm) [inline test/Microsoft.AspNetCore.Mvc.Razor.Test/Internal/ViewComponentTagHelperDescriptorFactoryTest.cs]: Could we split this out in to a separate test? `GetCSharpTypeName_ReturnsCorrectTypeNames_ForOutParameters`? This is kinda ugly

File: test/Microsoft.AspNetCore.Mvc.Razor.Host.Test/Microsoft.AspNetCore.Mvc.Razor.Host.Test.csproj [hunk 19]
```diff
@@ -0,0 +1,32 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <Import Project="..\..\build\common.props" />
+
+  <PropertyGroup>
+    <TargetFrameworks>netcoreapp1.1;net452</TargetFrameworks>
+    <TargetFrameworks Condition="'$(OS)' != 'Windows_NT'">netcoreapp1.1</TargetFrameworks>
+    <DefineConstants>$(DefineConstants);__RemoveThisBitTo__GENERATE_BASELINES</DefineConstants>
```
Comment (pranavkm) [inline test/Microsoft.AspNetCore.Mvc.Razor.Host.Test/Microsoft.AspNetCore.Mvc.Razor.Host.Test.csproj]:
```xml
<DefineConstants Condition="'$(GenerateBaselines)'=='true'">$(DefineConstants);GENERATE_BASELINES</DefineConstants>
<DefineConstants>$(DefineConstants);__RemoveThisBitTo__GENERATE_BASELINES</DefineConstants>
```

this lets you do `dotnet test /p:GenerateBaseLines:true` to produce baselines without editing any files. The second one exists as an option inside VS.

File: test/Microsoft.AspNetCore.Mvc.FunctionalTests/Microsoft.AspNetCore.Mvc.FunctionalTests.csproj (hunk 8 above)
Comment (pranavkm) [inline test/Microsoft.AspNetCore.Mvc.FunctionalTests/Microsoft.AspNetCore.Mvc.FunctionalTests.csproj]: We should probably add a tracking bug to re-add `net4x` testing. I forget why we decided to skip desktop here.

File: test/Microsoft.AspNetCore.Mvc.ApiExplorer.Test/Microsoft.AspNetCore.Mvc.ApiExplorer.Test.csproj (hunk 5 above)
Comment (pranavkm) [inline test/Microsoft.AspNetCore.Mvc.ApiExplorer.Test/Microsoft.AspNetCore.Mvc.ApiExplorer.Test.csproj]: I think Taylor tried this earlier, but we might have had newer packages since.

File: build/repo.props (hunk 17 above)
Comment (dougbu) [inline build/repo.props]: aspnet/KoreBuild#182

File: src/Microsoft.AspNetCore.Mvc.Core/Microsoft.AspNetCore.Mvc.Core.csproj (hunk 1 above)
Comment (dougbu) [inline src/Microsoft.AspNetCore.Mvc.Core/Microsoft.AspNetCore.Mvc.Core.csproj]: @pranavkm do you mean this should be `1.1.0` and not `1.1.0-*`?

File: test/Microsoft.AspNetCore.Mvc.FunctionalTests/Microsoft.AspNetCore.Mvc.FunctionalTests.csproj (hunk 7 above)
Comment (dougbu) [inline test/Microsoft.AspNetCore.Mvc.FunctionalTests/Microsoft.AspNetCore.Mvc.FunctionalTests.csproj]: See @pranavkm's [comment](https://github.com/aspnet/Mvc/pull/5852/files#r102830693) about the WebApiCompatShim project. #5822 should cover removing these last couple of fallbacks.

File: src/Microsoft.AspNetCore.Mvc.Core/Microsoft.AspNetCore.Mvc.Core.csproj (hunk 1 above)
Comment (pranavkm) [inline src/Microsoft.AspNetCore.Mvc.Core/Microsoft.AspNetCore.Mvc.Core.csproj]: Yup - the 1.1.0 package exists on nuget.org, so that's the build that'll get used: https://www.nuget.org/packages/Microsoft.Extensions.DependencyModel/1.1.0

File: test/WebSites/VersioningWebSite/VersioningWebSite.csproj [hunk 20]
```diff
@@ -0,0 +1,20 @@
+<Project Sdk="Microsoft.NET.Sdk.Web">
+
+  <Import Project="..\..\..\build\common.props" />
+
+  <PropertyGroup>
+    <TargetFrameworks>net452;netcoreapp1.1</TargetFrameworks>
+    <TargetFrameworks Condition="'$(OS)' != 'Windows_NT'">netcoreapp1.1</TargetFrameworks>
```
Comment (dougbu) [inline test/WebSites/VersioningWebSite/VersioningWebSite.csproj]: 🙈 not much point in compiling these and it adds at least 10s to a Linux build in my testing. Probably more since I only compared `build-compile` runs.

**Comment by dougbu** [comment]
:up::date:

Not sure why the AppVeyor build failed. Many functional tests hit issues in .NET 4.5.2 runs. Every failure I've seen was a `MissingMethodException` hit in `Microsoft.CodeAnalysis.Razor.TagHelperTypeVisitor.Create()` calls. Haven't reproduced this locally but am trying things out with all the latest packages&hellip;

**Comment by dougbu** [comment]
&hellip; bizarre but the [AppVeyor failures](https://ci.appveyor.com/project/aspnetci/mvc/build/1.0.8111) do not reproduce locally, even after `.\.build\nuget.exe locals all -clear` and `git clean -dfx`. And, the `dev` branch seems to [build fine](https://ci.appveyor.com/project/aspnetci/mvc/build/1.0.8112) as well. Redoing the [AppVeyor build](https://ci.appveyor.com/project/aspnetci/mvc/build/1.0.8113) of this PR&hellip;

**Comment by dougbu** [comment]
&hellip; latest [AppVeyor build](https://ci.appveyor.com/project/aspnetci/mvc/build/1.0.8113) of this PR failed in the same way. Will try to mimic AppVeyor environment locally&hellip;

**Comment by natemcmaster** [comment]
Talked with Doug in person. It appears the tests no longer pass when .NET Framework 4.6.2 is installed, which is causing the AppVeyor failures. We should look into this as it might be a bug in BCL or Roslyn.

**Comment by dougbu** [comment]
:up::date: FYI only. I squashed my changes since `nimullen/migration`.

**Comment by dougbu** [comment]
> It appears the tests no longer pass when .NET Framework 4.6.2 is installed, which is causing the AppVeyor failures.

I was wrong; .NET 4.6.2 _is_ installed on my machine. Must be some other difference between my machine and AppVeyor and @natemcmaster's machine.

**Comment by dougbu** [comment]
acfad83aa

**Comment by natemcmaster** [comment]
🎉
